### PR TITLE
Upsolve Pages Completed

### LINKED ISSUE #11: Upsolve Page Revamp

### PR DESCRIPTION
**Pull request by srijanjeet**
upsolve pages done. Solves #11

### CODE REVIEW

File: src/pages/Upsolve/upsolve.actions.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+export async function codeforces(accessToken, vir, page) {
```
Comment (Anurag5086) [inline src/pages/Upsolve/upsolve.actions.js]: Add the `upsolve.actions.js` page in the `/actions` folder in `/src` directory. Create a file name `Upsolve.js` and add all the API calls into this file.

File: src/styles/Upsolve/upsolve.css [hunk 2]
```diff
@@ -0,0 +1,297 @@
+*
+{
+    background-color : rgb(2 4 29);
+    /* background: linear-gradient(black,); */
```
Comment (Anurag5086) [inline src/styles/Upsolve/upsolve.css]: Remove comments!

File: src/styles/Upsolve/upsolve.css [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+    color: white;
+    border-radius: 10px;
+    height: 90px;
+    width: 19rem;
+    text-align: center;
+    padding-top: 10px;
+    box-shadow: 2px 2px 2px 3px black;
+  }
+  .upsolved {
+    /* background-color: rgb(0, 0, 0); */
```
Comment (Anurag5086) [inline src/styles/Upsolve/upsolve.css]: Remove all the comments in this file.

**Review by Anurag5086 — CHANGES_REQUESTED** [review]
Please make a upsolve component and make it common for all the 3 platforms and the upsolve pages dynamic using props. Don't make 3 separate components for all the 3 platforms.

File: src/styles/Upsolve/upsolve.css (hunk 3 above)
Comment (srijanjeet) [reply src/styles/Upsolve/upsolve.css]: done

File: src/styles/Upsolve/upsolve.css (hunk 2 above)
Comment (srijanjeet) [reply src/styles/Upsolve/upsolve.css]: Done

File: src/pages/Upsolve/upsolve.actions.js (hunk 1 above)
Comment (srijanjeet) [reply src/pages/Upsolve/upsolve.actions.js]: Done

File: src/actions/Upsolve.js [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+import { codeforces } from './upsolve.actions'
+import Validate from '../pages/Upsolve/Validate'
+
+export async function CodechefAPI(setFirst, setLast,
+    page,
+    setPrev,
+    setNext, setCurPage, setData, setLoader) {
+
+    const creds = JSON.parse(localStorage.getItem('creds'))
+    console.log(creds);
```
Comment (Anurag5086) [inline src/actions/Upsolve.js]: Remove

File: src/actions/Upsolve.js [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+export async function AtcoderAPI(setFirst, setLast,
+    page, Prac,
+    setPrev,
+    setNext, setCurPage, setData, setLoader) {
+    const creds = JSON.parse(localStorage.getItem('creds'))
+    const acc = creds.access
+    const response = await atcoder(acc, page, Prac)
+    if (response.status === 200) {
+        const data = await response.json()
+        //  console.log(data);
```
Comment (Anurag5086) [inline src/actions/Upsolve.js]: Remove

File: src/pages/Upsolve/CodeforcesUp.js [hunk 6]
```diff
@@ -1,12 +1,346 @@
...
+
+
+
+
+
+  return (
+    <>
+      {/* <Navbar></Navbar> */}
+      {/* <br></br>
+      <br></br>
```
Comment (Anurag5086) [inline src/pages/Upsolve/CodeforcesUp.js]: Remove `<br>`

File: src/pages/Upsolve/CodechefUp.js [hunk 7]
```diff
@@ -1,12 +1,330 @@
...
+import { Spin, Alert } from 'antd';
+import Carousel from 'react-multi-carousel'
+import { Row, Col } from 'antd';
+import "react-multi-carousel/lib/styles.css";
+import { Popover, Button } from 'antd';
+import { Switch } from 'antd';
+import Tags from '../../assets/Upsolve/tags-icon2.png'
+import logo from '../../assets/SitesImages/Codechef/codechef-png.png'
+import refresh from '../../assets/Upsolve/reload.png'
+import 'antd/dist/antd.css';
```
Comment (Anurag5086) [inline src/pages/Upsolve/CodechefUp.js]: Remove this

**Review by Anurag5086 — CHANGES_REQUESTED** [review]
Do the suggested changes!

File: src/pages/Upsolve/Toggle.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+                    <div>
+                        <button className='btnUpdate'
+                            title="solved? update"
+                            
+                            onClick={(e) => {
+                                props.setUpdate(props.update + 1)
+                            }}
+                        >
+                            <img
+                                style={{ width: '50px', height: '52px' }}
```
Comment (Anurag5086) [inline src/pages/Upsolve/Toggle.js]: Remove inline css

File: src/pages/Upsolve/AtcoderUp.js [hunk 9]
```diff
@@ -1,12 +1,97 @@
...
+            <br></br>
+            {loader ? (
+                // <Spinner className="loading-animation" animation="border" />
+                <p>hello</p>
+            ) : (
+                <div className="body">
+                    {conData.length > 0 ? (
+                        <>
+                            <Toggle siteName="AtCoder" update = {update} logo={logo} imgWidth="fit-content" imgHeight="50px" setUpdate={setUpdate} refresh={refresh} wn={wn} setWN={setWN} ChangePage={ChangePage} />
+                            <br></br>
```
Comment (Anurag5086) [inline src/pages/Upsolve/AtcoderUp.js]: Remove `<br>`, instead use css to add spacing between elements.

File: src/pages/Upsolve/AtcoderUp.js [hunk 10]
```diff
@@ -1,12 +1,97 @@
...
+                                )
+                            })}
+                            <div className="paginate">
+                            <Lastpages page = {page} setLoader = {setLoader} pageNumbers = {pageNumbers} setPage = {setPage} setCurPage = {setCurPage} last = {last}/>
+                            </div>
+
+                            {/* <Footer /> */}
+                        </>
+                    ) : (
+                        <p>hello</p>
```
Comment (Anurag5086) [inline src/pages/Upsolve/AtcoderUp.js]: Change `hello` to `Loading` for now.

File: src/actions/Upsolve.js [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+import { codeforces } from './upsolve.actions'
+import Validate from '../pages/Upsolve/Validate'
+
+export async function CodechefAPI(setFirst, setLast,
+    page,
+    setPrev,
+    setNext, setCurPage, setData, setLoader) {
+
+    const creds = JSON.parse(localStorage.getItem('creds'))
+    const acc = creds.access
```
Comment (Anurag5086) [inline src/actions/Upsolve.js]: Instead of `acc` use `accessToken` as the variable name

File: src/pages/Upsolve/AtcoderUp.js [hunk 12]
```diff
@@ -1,12 +1,97 @@
...
-    console.log(props);
+const Atcoder = () => {
+    const pageNumbers = []
+    const [page, setPage] = useState(1)
+    const [loader, setLoader] = useState(false)
+    const [prev, setPrev] = useState(null)
+    const [next, setNext] = useState(2)
+    const [first, setFirst] = useState(1)
+    const [last, setLast] = useState(null)
+    const [conData, setData] = useState([])
```
Comment (Anurag5086) [inline src/pages/Upsolve/AtcoderUp.js]: Use understandable variables. I cant understand what `conData` means. So try to use complete words as vairable, dosen't matter if they are long.

File: src/pages/Upsolve/AtcoderUp.js [hunk 13]
```diff
@@ -1,12 +1,97 @@
...
+import Lastpages from './Lastpages';
 
-function AtcoderUp(props) {
-    console.log(props);
+const Atcoder = () => {
+    const pageNumbers = []
+    const [page, setPage] = useState(1)
+    const [loader, setLoader] = useState(false)
+    const [prev, setPrev] = useState(null)
+    const [next, setNext] = useState(2)
```
Comment (Anurag5086) [inline src/pages/Upsolve/AtcoderUp.js]: Change the name here as well, `nextPage` or something of this sort.

File: src/pages/Upsolve/CarouselUpsolve.js [hunk 14]
```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import { Row, Col } from 'antd';
+import Carousel from 'react-multi-carousel'
+import "react-multi-carousel/lib/styles.css";
```
Comment (Anurag5086) [inline src/pages/Upsolve/CarouselUpsolve.js]: All the css imports should be written below the Js imports.

File: src/pages/Upsolve/CodechefUp.js [hunk 15]
```diff
@@ -1,12 +1,105 @@
...
+import logo from '../../assets/SitesImages/Codechef/codechef-png.png'
+import refresh from '../../assets/Upsolve/reload.png'
+import { CodechefAPI } from '../../actions/Upsolve';
+import Toggle from './Toggle';
+import Carousel_Upsolve from './CarouselUpsolve';
+import Lastpages from './Lastpages';
+
+const Codechef = () => {
+  let [update, setUpdate] = useState(0)
+  const [wn, setWN] = useState(false)
```
Comment (Anurag5086) [inline src/pages/Upsolve/CodechefUp.js]: Add meaningful and understandable variables. `wn` is not understandable.

**Review by Anurag5086 — CHANGES_REQUESTED** [review]
Do the requested changes.

**Comment by Anurag5086** [comment]
@srijanjeet Also resolve the conflicting files.

File: src/pages/Upsolve/AtcoderUp.js (hunk 10 above)
Comment (srijanjeet) [reply src/pages/Upsolve/AtcoderUp.js]: done

File: src/pages/Upsolve/Toggle.js (hunk 8 above)
Comment (srijanjeet) [reply src/pages/Upsolve/Toggle.js]: done

File: src/pages/Upsolve/AtcoderUp.js (hunk 9 above)
Comment (srijanjeet) [reply src/pages/Upsolve/AtcoderUp.js]: done

File: src/pages/Upsolve/CarouselUpsolve.js (hunk 14 above)
Comment (srijanjeet) [reply src/pages/Upsolve/CarouselUpsolve.js]: done

**Review by Anurag5086 — CHANGES_REQUESTED** [review]
Remove comments

**Review by Anurag5086 — APPROVED** [review]
LGTM

**Pull request merged**